test/hotspot/jtreg/compiler/lib/generators/Generators.java

```diff
@@ -165,13 +166,35 @@ public Generator<Double> anyBitsDouble() {
         return new AnyBitsDoubleGenerator(this);
     }
 
+    /**
+     * Generates uniform float16s in the range of [lo, hi) (inclusive of lo, exclusive of hi).
+     */
+    public RestrictableGenerator<Short> uniformFloat16s(short lo, short hi) {
```

Comment:
Hmm, passing `short` into this API seems a little strange to me. Because the range `[lo, hi)` doesn't really make sense... rather it is the bits by `lo` and `hi` interpreted as `Float16`... but isn't that a little cumbersome? Why not pass in `Float16` instead?

Reply:
We want to avoid making a reference to an incubating class for now as every test using generator will then need to --add-modules=jdk.incubator.vector. Also, this will not pose any issue when we add a new Generator<Short> for short values since underlines framework APIs will be different from the once used for float16 generator with Short carrier type.